### PR TITLE
Update app to latest Node version

### PR DESCRIPTION
**Pull request by thecalliopecrashed**
This change includes several different commits;
walk through the commits individually for details on changes.

There is no net functional change in the app;
instead, this change improves the overal robustness of the app
and speeds up the developer feedback loop with automatic reloading.

### CODE REVIEW

**Comment by thecalliopecrashed** [comment]
Demos that:

1. Functionality remains unchanged

functionality

2. Development experience is improved

dev_loop

**Comment by thecalliopecrashed** [comment]
@pmanko Can you run through the following test script on your computer for me?

```bash
type node
type yarn
node -v
yarn -v

# in your local copy of the `tb-mobile-app` repo
git checkout uptodate

# This might break if your node version < 10
yarn
yarn start
```

A browser page should open for you.
If you make a change to `src/App.js`,
you should see the change appear in the UI.

Now, to test that it works properly in Docker: (Make sure to replace the paths!)

```
cd ~/path_to/tb-api

git pull
echo "COMPOSE_FILE=docker-compose.yml:docker-compose.dev.yml" >> .env
echo "TB_MOBILE_CONTEXT=~/path_to/tb-mobile-app"
docker-compose up -d
```

Click around the app, make sure everything works.

If you change `src/App.js` again, you should see the page pick up the changes.

Let me know how it looks! Thanks.

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+
+it('renders without crashing', () => {
```
Comment (ivan-c) [inline src/App.test.js]: Are the tests passing? How is the test suite started? (thinking about setting up some CI)

Comment (thecalliopecrashed) [reply src/App.test.js]: As far as I can tell, all the tests are failing.

I haven't looked at the test cases much, but I'd like to get them running again.
I'll take a look when I start moving the logout button.

You should be able to run `yarn test`; you'll get an interactive command-line test runner.

Any preferences for a CI platform? I can start setting that up.

**Comment by thecalliopecrashed** [comment]
> (node:8) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.

Ah right, I saw that too.
It looks like this is coming from `yarn`, and was [fixed in a more recent version](https://github.com/yarnpkg/yarn/issues/5477#issuecomment-412064714).

I'll see if I can upgrade our yarn version.

File: src/App.test.js (hunk 1 above)
Comment (ivan-c) [reply src/App.test.js]: Thanks for the explanation, glad to hear you have a rough plan to get tests working again.

>Any preferences for a CI platform? I can start setting that up.

I have the most experience with TravisCI, and found it pretty flexible (root access and docker support and great free plan for FOSS), but I'm not attached if there's something you're more comfortable with.

**Pull request merged**